python/tk_multi_workfiles/browser_form.py

```diff
@@ -705,11 +709,183 @@ def _on_task_status_combo_changed(self, value):
+        )
+        fields_manager.initialized.connect(self._field_filters)
+        fields_manager.initialize()
+        self._sort_menu_actions()
+
+    def _field_filters(self):
+        """
+        Define a few simple filter methods for use by the menu
+        """
+
+        def field_filter(field):
+            # none of the fields are included
+            return False
+
+        def checked_filter(field):
+            # none of the fields are checked
+            return False
+
+        def disabled_filter(field):
+            # none of the fields are disabled
+            return False
+
+        # attach the filters
+        self._entity_field_menu.set_field_filter(field_filter)
+        self._entity_field_menu.set_checked_filter(checked_filter)
+        self._entity_field_menu.set_disabled_filter(disabled_filter)
+
+    def _sort_menu_actions(self, tab_name="tasks"):
+        """
+        Populate the sort menu with actions.
+        """
+        # Get settings from info.yml
+        app = sgtk.platform.current_bundle()
+        sort_fields = app.get_setting("sort_fields").get(tab_name, [])
+
+        # Create Sort Menu actions
+        sort_asc = self._entity_field_menu._get_qaction("ascending", "Ascending")
+        sort_desc = self._entity_field_menu._get_qaction("descending", "Descending")
+        separator = self._entity_field_menu.addSeparator()
+        field_sort_actions = [
+            self._entity_field_menu._get_qaction(
+                field["field_code"], field["display_name"]
+            )
+            for field in sort_fields or []
+        ]
+
+        # Actions group list ordered
+        sort_actions = [sort_asc, sort_desc, separator, *field_sort_actions]
+
+        # By default it sorts in descending order and the default field is set in the configuration
+        sort_desc.setChecked(True)
+
+        # Menu sort order actions
+        sort_asc.triggered[()].connect(
+            lambda: self.load_sort_data(
+                "ascending", sort_asc, sort_actions, sort_order="asc"
+            )
+        )
+        sort_desc.triggered[()].connect(
+            lambda: self.load_sort_data(
+                "descending", sort_desc, sort_actions, sort_order="desc"
+            )
+        )
+        # Menu sort field actions
+        # default_set = False
+        for index, field_sort_action in enumerate(field_sort_actions):
+            sort_field = sort_fields[index]
+            field_code = sort_field["field_code"]
+            field_sort_action.triggered[()].connect(
+                lambda fc=field_code, fsa=field_sort_action: self.load_sort_data(
+                    fc, fsa, sort_actions
+                )
+            )
+
+        # Add actions to the entity Menu
+        self._entity_field_menu.add_group(sort_actions, "Sort menu")
+        # Remove the separator from the list
+        sort_actions.remove(separator)
+
+    def load_sort_data(self, field, sort_action, actions_list, **sort_order):
```

Comment:
I don't understand this syntax. Do you mean you want `sort_order` to be a mandatory named parameter? If that's so, I believe the syntax is:

```suggestion
    def load_sort_data(self, field, sort_action, actions_list, *, sort_order):
```

What did I miss?

Reply:
It shouldn't be mandatory, otherwise it will fail on line 826. 

The invocation actually looks like this: `sort_order="desc"`

I think it would be best if I rename this to `**kwargs` or just `sort_order="asc"`.